File: src/assignment3/src/plan_validator_node.cpp
#include "ros/ros.h"
#include "assignment3/ValidatePlan.h"

bool validate_plan(assignment3::ValidatePlan::Request &req, assignment3::ValidatePlan::Response &res) {
    // Validate the current plan based on the state
    res.valid = true; // Replace with actual validation logic
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "plan_validator_node");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("validate_plan", validate_plan);
    ros::spin();

    return 0;
}

File: src/assignment3/src/plan_validator_node.py
import rospy
from assignment3.srv import ValidatePlan, ValidatePlanResponse

def validate_plan(req):
    # Validate the current plan based on the state
    valid = True  # Replace with actual validation logic
    return ValidatePlanResponse(valid)

def plan_validator_server():
    rospy.init_node('plan_validator_node')
    service = rospy.Service('validate_plan', ValidatePlan, validate_plan)
    rospy.spin()

if __name__ == "__main__":
    plan_validator_server()
